search/includes/classes/class-search.php

```diff
@@ -902,15 +902,21 @@ public function filter__ep_do_intercept_request( $request, $query, $args, $failu
 		$statsd_prefix          = $this->get_statsd_prefix( $query['url'], $statsd_mode );
 
 		// Cache handling
-		$is_cacheable = $this->is_url_query_cacheable( $query['url'], $args );
-		$cache_key    = 'es_query_cache:' . md5( $query['url'] . wp_json_encode( $args ) ) . ':' . wp_cache_get_last_changed( self::SEARCH_CACHE_GROUP );
+		$is_cacheable    = $this->is_url_query_cacheable( $query['url'], $args );
+		$cached_response = false;
 
-		/**
-		 * Serve cached response right away, if available and not stale and the query is cacheable
-		 */
-		$cached_response = $is_cacheable ? wp_cache_get( $cache_key, self::SEARCH_CACHE_GROUP ) : false;
-		if ( $cached_response ) {
-			return $cached_response;
+		if ( $is_cacheable ) {
+			// Some requests may not have body
+			$body      = $args['body'] ?? '';
+			$cache_key = 'es_query_cache:' . md5( $query['url'] . $body ) . ':' . wp_cache_get_last_changed( self::SEARCH_CACHE_GROUP );
```

Comment:
Since `$cache_key` is only defined in this block here, will it be a `null` value when it comes to line 1000 at `wp_cache_set( $cache_key, $response, self::SEARCH_CACHE_GROUP, 5 * MINUTE_IN_SECONDS );`?

Reply:
It's wrapped in the similar `$is_cacheable` check so I think it should be fine?